.github/workflows/release.yaml

```diff
@@ -1,63 +1,78 @@
 name: Release
 
 on:
-  push:
-    tags:
-      # we don't want to be triggered for RC releases (e.g. v0.6.0-rc9)
-      - 'v[0-9]+.[0-9]+.[0-9]+'
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'The version to release, without prefix v (e.g. 1.1.0)'
+        required: true
+      branch:
+        description: 'The branch to release from (e.g. 1.0.x)'
+        required: true
+        default: 'master'
 
 jobs:
-  push-to-release-branch:
-    name: Push source to release branch
+  trigger-and-wait-for-mlrun-image-building:
+    name: Trigger build workflow in mlrun/mlrun and wait to finish
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-      with:
-        # we can't use the default token coming with the action cause GH prevents from create or updating workflows
-        # which might be part of the release contents
-        token: ${{ secrets.RELEASE_GITHUB_ACCESS_TOKEN }}
-    - name: Push to release branch
-      run: |
-        export tag=$(echo ${GITHUB_REF#refs/tags/})
-        git config user.name "mlrun-iguazio"
-        git config user.email "mlrun.iguazio@gmail.com"
-        MLRUN_GIT_ORG=${{ github.repository_owner }} MLRUN_VERSION=$tag make release
-
+      - uses: convictional/trigger-workflow-and-wait@v1.6.3
+        with:
+          owner: mlrun
+          repo: mlrun
+          github_token: ${{ secrets.RELEASE_GITHUB_ACCESS_TOKEN }}
+          workflow_file_name: build.yaml
+          ref: ${{ github.event.inputs.branch }}
+          wait_interval: 60
+          client_payload: '{"docker_registries": "ghcr.io/,quay.io/,registry.hub.docker.com/", "version": "${{ github.event.inputs.version }}"}'
 
-  trigger-build-workflow:
-    # When the build workflow is triggered for a commit to development it builds the images and (docker) tag them with
-    # <latest-released-version>-<commit-hash>.
-    # When the periodic system tests workflow runs, it gets the latest commit from development, and searches for the
-    # relevant tag.
-    # Therefore, when a new version is released, any system test that will run after it, and before a new commit to
-    # development was done, will search for a docker tag that doesn't exist.
-    # In order to solve that we trigger the build workflow on the development branch when we release a new version
-    name: Trigger build workflow on development branch
+  trigger-and-wait-for-ui-image-building:
+    name: Trigger build workflow in mlrun/ui and wait to finish
     runs-on: ubuntu-latest
 
     steps:
+      - uses: convictional/trigger-workflow-and-wait@v1.6.3
+        with:
+          owner: mlrun
+          repo: ui
+          github_token: ${{ secrets.RELEASE_GITHUB_ACCESS_TOKEN }}
+          workflow_file_name: build.yaml
+          ref: ${{ github.event.inputs.branch }}
+          wait_interval: 60
+          client_payload: '{"docker_registries": "ghcr.io/,quay.io/,registry.hub.docker.com/", "version": "${{ github.event.inputs.version }}"}'
```

Comment:
How this is expected to handle failing workflows?

Reply:
IIUC the timeout is only if the workflow it triggered takes long time, if it fails in the mean time and then runs successfully on the second time still in the period of the interval, this workflow will fail?